alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -670,41 +671,67 @@ object MonitorRunner : JobRunner, CoroutineScope, AbstractLifecycleComponent() {
         }
     }
 
-    // TODO: This is largely a duplicate of runAction above for BucketLevelTriggerExecutionContext for now.
-    //   After suppression logic implementation, if this remains mostly the same, it can be refactored.
-    private suspend fun runAction(action: Action, ctx: BucketLevelTriggerExecutionContext, dryrun: Boolean): ActionRunResult {
-        return try {
-            val actionOutput = mutableMapOf<String, String>()
-            actionOutput[SUBJECT] = if (action.subjectTemplate != null) compileTemplate(action.subjectTemplate, ctx) else ""
-            actionOutput[MESSAGE] = compileTemplate(action.messageTemplate, ctx)
-            if (Strings.isNullOrEmpty(actionOutput[MESSAGE])) {
-                throw IllegalStateException("Message content missing in the Destination with id: ${action.destinationId}")
-            }
-            if (!dryrun) {
-                withContext(Dispatchers.IO) {
-                    val destination = AlertingConfigAccessor.getDestinationInfo(client, xContentRegistry, action.destinationId)
-                    if (!destination.isAllowed(allowList)) {
-                        throw IllegalStateException("Monitor contains a Destination type that is not allowed: ${destination.type}")
-                    }
+    private suspend fun getConfigAndSendNotification(action: Action, subject: String?, message: String): String {
+        val config = getConfigForNotificationAction(action)
 
-                    val destinationCtx = destinationContextFactory.getDestinationContext(destination)
-                    actionOutput[MESSAGE_ID] = destination.publish(
-                        actionOutput[SUBJECT],
-                        actionOutput[MESSAGE]!!,
-                        destinationCtx,
-                        hostDenyList
-                    )
-                }
-            }
-            ActionRunResult(action.id, action.name, actionOutput, false, currentTime(), null)
-        } catch (e: Exception) {
-            ActionRunResult(action.id, action.name, mapOf(), false, currentTime(), e)
+        if (config.destination == null && config.channel == null) {
+            throw IllegalStateException("Unable to find a Notification Channel or Destination config with id [${action.id}]")
+        }
+
+        // Adding a check on TEST_ACTION Destination type here to avoid supporting it as a LegacyBaseMessage type
+        // just for Alerting integration tests
+        if (config.destination?.isTestAction() == true) {
+            return "test action"
```

Comment:
can we define this constant somewhere else and use it both at the time assert and here.

Reply:
+1